examples/siro_sandbox/controllers/fetch_baselines_controller.py

```diff
@@ -410,10 +420,79 @@ def act(self, obs, env):
         if self._beg_state_lock:
             self.current_state = FetchState.BEG_RESET
 
-        if self.current_state == FetchState.SEARCH:
+        if self.current_state == FetchState.WAIT:
+            self.current_state = FetchState.FOLLOW
+            self._init_policy_input()
+        elif self.current_state == FetchState.FOLLOW:
+            # This is the following state, in which the robot tries to follow the human
+            # There are two cases here. When the human is far away from the robot and robot cannot see the human,
+            # we use normal point nav, and when the human is close enough or robot can see the human,
+            # we switch to the social nav. This allows us to effectively find and follow the human
+            # First get the geo distance between the robot and the human and check if the robot can see human
+            human_robot_geo_dis = self._habitat_env._sim.geodesic_distance(
+                robot_pos, human_pos
+            )
+            human_in_frame = self._batch_and_apply_transforms([obs])[
+                "humanoid_detector_sensor"
+            ]
+            if (
+                human_robot_geo_dis
+                > FOLLOW_SWITCH_GEO_DIS_FOR_POINT_SOCIAL_NAV
+                and not bool(human_in_frame)
```

Comment:
This heuristic is fine for this PR, but let's plan to discuss this in an upcoming HITL meeting. As a demo presenter, I usually like stuff like this to be very transparent and explainable to the audience. I see you have a switch to enable/disable this, which is great!

Reply:
taking note here, and I put the comment in the Asana. thank you! Eric!